### PR TITLE
DAOS-6540 dfuse: Use correct uid/gid bits when creating files.

### PR DESCRIPTION
**Pull request by ashleypittman**
Pass in the required uid/gid to dfs when creating either files
or directories rather than having dfs call getuid().

This allows groups to work properly in dfuse, and will also
support multi-user dfuse when enabled.

Signed-off-by: Ashley Pittman <ashley.m.pittman@intel.com>

### CODE REVIEW

File: src/client/dfs/dfs_internal.h [hunk 1]
```diff
@@ -66,7 +66,11 @@ dfs_get_sb_layout(daos_key_t *dkey, daos_iod_t *iods[], int *akey_count,
 void
 dfs_free_sb_layout(daos_iod_t *iods[]);
 
-/** as dfs_open() but takes a stbuf to be populated if O_CREATE is specified */
+/** as dfs_open() but takes a stbuf to be populated if O_CREATE is specified
+ *
+ * If O_CREATE is set then entry->uid and entry->gid are popouated with the desired user,
```
Comment (mchaarawi) [inline src/client/dfs/dfs_internal.h:71]: i am fine with change as this is an internal API.

File: src/tests/suite/dfs_unit_test.c [hunk 2]
```diff
@@ -393,6 +393,56 @@ dfs_test_lookup(void **state)
 	assert_int_equal(rc, 0);
 }
 
+static void
+dfs_test_open_stat(void **state)
```
Comment (mchaarawi) [inline src/tests/suite/dfs_unit_test.c:397]: instead of adding a new test, please just extend the dfs_test_chown to test also with open_stat.

Comment (mchaarawi) [reply src/tests/suite/dfs_unit_test.c:397]: should probably also add an NLT test for dfuse level test

File: src/client/dfs/dfs_internal.h (hunk 1 above)
Comment (ashleypittman) [reply src/client/dfs/dfs_internal.h:71]: Yes, it's not a great interface but this is why it's internal.

File: src/tests/suite/dfs_unit_test.c (hunk 2 above)
Comment (ashleypittman) [reply src/tests/suite/dfs_unit_test.c:397]: Will do.

I've tested dfuse with this code (it's a port of some of the multi-user code) but as yet you can't call it from dfuse yet as it rejects all I/O if the user/group doesn't match entirely.  This PR has the effect of changing how dfuse sets the uid/gid so it's set from the kernel request not get*id in the dfs code, but it is not enough to allow dfuse to set different values.

Comment (ashleypittman) [reply src/tests/suite/dfs_unit_test.c]: Done.

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**